src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/UpdateWorkitem.sql

```diff
@@ -0,0 +1,211 @@
+		        wi.WorkitemKey,
+		        wi.WorkitemUid,
+		        wi.TransactionUid,
+		        eqts.Watermark
+	        FROM 
+		        dbo.WorkitemQueryTag wqt
+                INNER JOIN @dateTimeExtendedQueryTags input
+                    ON input.TagKey = wqt.TagKey
+		        INNER JOIN dbo.ExtendedQueryTagDateTime eqts ON 
+			        eqts.TagKey = wqt.TagKey 
+			        AND eqts.ResourceType = 1 -- Workitem Resource Type
+		        INNER JOIN dbo.Workitem wi ON
+			        wi.PartitionKey = eqts.PartitionKey
+			        AND wi.WorkitemKey = eqts.SopInstanceKey1
+	        WHERE
+		        wi.PartitionKey = @partitionKey
+		        AND wi.WorkitemKey = @workitemKey
+        )
+        UPDATE targetTbl
+        SET
+            targetTbl.TagValue = cte.NewTagValue,
+            targetTbl.Watermark = @newWatermark
+        FROM
+	        dbo.ExtendedQueryTagDateTime targetTbl
+	        INNER JOIN InputCTE cte ON
+		        targetTbl.ResourceType = cte.ResourceType
+		        AND cte.PartitionKey = targetTbl.PartitionKey
+		        AND cte.WorkitemKey = targetTbl.SopInstanceKey1
+		        AND cte.TagKey = targetTbl.TagKey
+		        AND cte.OldTagValue = targetTbl.TagValue
+		        AND cte.Watermark = targetTbl.Watermark
+
+    END
+
+    -- PersonName Key tags
+    IF EXISTS (SELECT 1 FROM @personNameExtendedQueryTags)
+    BEGIN
+
+        WITH InputCTE AS (
+	        SELECT
+		        wqt.TagKey,
+		        wqt.TagPath,
+		        eqts.TagValue AS OldTagValue,
+                input.TagValue AS NewTagValue,
+                eqts.ResourceType,
+		        wi.PartitionKey,
+		        wi.WorkitemKey,
+		        wi.WorkitemUid,
+		        wi.TransactionUid,
+		        eqts.Watermark
+	        FROM 
+		        dbo.WorkitemQueryTag wqt
+                INNER JOIN @personNameExtendedQueryTags input
+                    ON input.TagKey = wqt.TagKey
+		        INNER JOIN dbo.ExtendedQueryTagPersonName eqts ON 
+			        eqts.TagKey = wqt.TagKey 
+			        AND eqts.ResourceType = 1 -- Workitem Resource Type
+		        INNER JOIN dbo.Workitem wi ON
+			        wi.PartitionKey = eqts.PartitionKey
+			        AND wi.WorkitemKey = eqts.SopInstanceKey1
+	        WHERE
+		        wi.PartitionKey = @partitionKey
+		        AND wi.WorkitemKey = @workitemKey
+        )
+        UPDATE targetTbl
+        SET
+            targetTbl.TagValue = cte.NewTagValue,
+            targetTbl.Watermark = @newWatermark
+        FROM
+	        dbo.ExtendedQueryTagPersonName targetTbl
+	        INNER JOIN InputCTE cte ON
+		        targetTbl.ResourceType = cte.ResourceType
+		        AND cte.PartitionKey = targetTbl.PartitionKey
+		        AND cte.WorkitemKey = targetTbl.SopInstanceKey1
+		        AND cte.TagKey = targetTbl.TagKey
+		        AND cte.OldTagValue = targetTbl.TagValue
+		        AND cte.Watermark = targetTbl.Watermark
+
+    END
+
+    -- Update the Workitem status
+    UPDATE dbo.Workitem
+    SET
+        [Status] = @status,
```

Comment:
It feels weird we are doing this update here, either we should call that as separate step or call that sproc here

Reply:
I'll change this to call "dbo.UpdateWorkitemStatus" but would still want to keep it under this transaction.